### PR TITLE
Feature #56: Initialize the React Native Project

### PR DESCRIPTION
**Pull request by TienDucHo**
In this pull request, I have done the following:

- Initialize the React Native Project
- Setting up prettier and eslint
- Cleaning up boilerplate code in App.js
- Added example test using Jest
- Added README.md that includes Installation, Running and Testing instruction

@dimassudjito please review.

### CODE REVIEW

File: frontend/ICard/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# ICard Mobile Application
```
Comment (dimassudjito) [inline frontend/ICard/README.md:1]: Can update running instruction in README.md to use npm shortcut: npm start, npm run android, npm run ios, npm test

File: frontend/ICard/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+## Running Application
+
+Type in terminal to start Metro
+
+```bash
+npx react-native start
+```
+
+### Android
```
Comment (dimassudjito) [inline frontend/ICard/README.md:19]: Add in README.md instructions to clean gradle when running apps in android https://stackoverflow.com/questions/42570067/react-native-always-gradlew-clean-before-run-react-native-run-android

File: frontend/ICard/.eslintrc.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 module.exports = {
   root: true,
   extends: '@react-native-community',
+  rules: {'prettier/prettier': ['error', {endOfLine: 'auto'}]},
```
Comment (dimassudjito) [inline frontend/ICard/.eslintrc.js:4]: I see prettier is working nicely, but I don’t see it installed in package.json. Will this work on a machine without prettier’s extension installed? (I have the VS Code extension in my machine)

Comment (dimassudjito) [reply frontend/ICard/.eslintrc.js:4]: If installing prettier, consider adding eslint-plugin-prettier to make prettier works nicely with eslint

**Review by dimassudjito — CHANGES_REQUESTED** [review]
- Move all content of “frontend/ICard” to a “mobile/” folder directly under the ICard repo
- What do you think of a good project structure would be? I'm thinking of having having the following structure
  - mobile
    - android/
    - ios/
    - src/
      - app.js
      - index.js
      - components/ (component used in pages)
      - views/ (pages)
      - routes/

File: frontend/ICard/.eslintrc.js (hunk 3 above)
Comment (TienDucHo) [reply frontend/ICard/.eslintrc.js:4]: I haven't test without prettier extension, I will include prettier inside package.json. Will further include eslint-plugin-prettier to the installation too

**Review by TienDucHo — COMMENTED** [review]
Create new structure as requested

File: frontend/ICard/.eslintrc.js (hunk 3 above)
Comment (TienDucHo) [reply frontend/ICard/.eslintrc.js:4]: Fixed with https://github.com/UofA-Blueprint/ICard/pull/1/commits/138a50024e8e2a778eb948c65f80199aadbfb309

File: frontend/ICard/README.md (hunk 2 above)
Comment (TienDucHo) [reply frontend/ICard/README.md]: Added with https://github.com/UofA-Blueprint/ICard/pull/1/commits/612552850bce1a529d2b705c2b8db5e583465a68

File: frontend/ICard/README.md (hunk 1 above)
Comment (TienDucHo) [reply frontend/ICard/README.md]: Added with https://github.com/UofA-Blueprint/ICard/pull/1/commits/612552850bce1a529d2b705c2b8db5e583465a68

**Comment by dimassudjito** [comment]
@TienDucHo 

- I think you accidentally installed prettier and eslint-plugin-prettier in the root folder so it created a new package.json and node_modules folder which are tracked on git (they should be installed on the mobile/ folder)
- git don't really  show empty folder, so if you want to create a skeleton for the project structure (src folder containing components, views, etc), you probably need to add empty .js/.jsx file 
- I kinda regret not using .jsx file in uofablueprint.org from the start, is it possible to use .jsx file for the react file in react native?

**Comment by TienDucHo** [comment]
> 

Fixed with https://github.com/UofA-Blueprint/ICard/pull/1/commits/61b3f5480bf229d2d9938456f3267c95b0b495d3
I have configure our project to use both js and jsx file. I have some example files inside so you can test.

**Comment by TienDucHo** [comment]
> @TienDucHo
> 
> * I think you accidentally installed prettier and eslint-plugin-prettier in the root folder so it created a new package.json and node_modules folder which are tracked on git (they should be installed on the mobile/ folder)
> * git don't really  show empty folder, so if you want to create a skeleton for the project structure (src folder containing components, views, etc), you probably need to add empty .js/.jsx file
> * I kinda regret not using .jsx file in uofablueprint.org from the start, is it possible to use .jsx file for the react file in react native?

Fixed the accidentally installed npm modules file in root with https://github.com/UofA-Blueprint/ICard/commit/d41faacf37788f0a5362a9f8e75200304cd610d6

**Comment by dimassudjito** [comment]
> > @TienDucHo
> > 
> > * I think you accidentally installed prettier and eslint-plugin-prettier in the root folder so it created a new package.json and node_modules folder which are tracked on git (they should be installed on the mobile/ folder)
> > * git don't really  show empty folder, so if you want to create a skeleton for the project structure (src folder containing components, views, etc), you probably need to add empty .js/.jsx file
> > * I kinda regret not using .jsx file in uofablueprint.org from the start, is it possible to use .jsx file for the react file in react native?
> 
> Fixed the accidentally installed npm modules file in root with [d41faac](https://github.com/UofA-Blueprint/ICard/commit/d41faacf37788f0a5362a9f8e75200304cd610d6)

@TienDucHo  Awesome! One last thing, could we install prettier and eslint-prettier-plugin as devDependencies instead of dependencies? We can definitely merge after that. Really good job on the project setup, thanks!

**Comment by TienDucHo** [comment]
> Update the dependencies with https://github.com/UofA-Blueprint/ICard/pull/1/commits/9aa878ae46ea2963fcd7e6beea736e2d97dd9baa

**Pull request merged**